### PR TITLE
TD-2286 replace the existing tracking system support support tickets page with a request support page 

### PR DESCRIPTION
**Pull request by sarathlal-sarangadharan**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-2286

### Description
Existing Tracking System / Support / Support tickets page has been replaced  with a Request support page

### Screenshots
image
image
image
image
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [ ] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            this.connection = connection;
+        }
+        public IEnumerable<RequestType> GetRequestTypes()
+        {
+            return connection.Query<RequestType>(@$"SELECT TicketTypeId as ID,TypePrompt AS RequestTypes,FreshdeskTicketType AS FreshdeskRequestTypes FROM [dbo].[TicketTypes]  order by TypePrompt");
+        }
+        public string? GetUserCentreEmail(int userId, int centreId)
+        {
+            //return connection.QuerySingleOrDefault<string>(
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs:28]: Do we have any purpose for this commented query?

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                        <dd class="nhsuk-summary-list__value">
+                            @Model.Description
+                        </dd>
+                        <dd class="nhsuk-summary-list__actions">
+
+                        </dd>
+                    </div>
+                    <div class="nhsuk-summary-list__row">
+                        <dt class="nhsuk-summary-list__key">
+                            Attachements
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml:59]: Correct the spelling of this.

**Review by rshrirohit — CHANGES_REQUESTED** [review]
1. The menu at the top in blue ribbon is not showing other options - 
image

2. Redirect this link as well to the newly created functionality instead of IFrame.
image

3. The heading of this page should match the previously selected radio button label - 
image


4. OR separator is missing as per ticket description
image


5. Required Validation missing on description text area
6. Validation on file size is not working properly
image


7. Redirecting from the summary page to the attachments page via clicking the Change hyperlink shows a validation error even though the field is not mandatory.
image

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+<div id="form-group">
+    <div class="nhsuk-grid-row">
+        <div class="nhsuk-grid-column-full">
+            <div class="nhsuk-u-reading-width">
+                <div class="nhsuk-u-margin-bottom-8">
+                    <h1 class="nhsuk-heading-xl">Attach screenshots (optional)</h1>
+                </div>
+                <form class="nhsuk-u-margin-bottom-3" method="post" asp-action="SetAttachment" enctype="multipart/form-data">
+                    <p class="nhsuk-body-s nhsuk-u-margin-bottom-3">
+                        If you have any screenshots that help illustarte your request,attach them here.
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml:31]: correct the spelling of illustrate

File: DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs (hunk 1 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs:28]: Hi @rshrirohit , I had a doubt on this and added in ticket, @kevwhitt-hee  please have a look. 
image

Left the query commented to use if required.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        errorHasOccurred = true;
+    }
+
+}
+@if (errorHasOccurred )
+{
+    <vc:error-summary order-of-property-names="@(new[] { nameof(Model.ImageFiles),nameof(Model.FileSizeError),nameof(Model.FileExtensionError) })" />
+}
+<a class="nhsuk-button nhsuk-button--secondary" role="button" asp-controller="RequestSupportTicket" asp-action="RequestSummary">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml:19]: These Back links should use the [NHS Design System back-link component](https://service-manual.nhs.uk/design-system/components/back-link), as per wireframes.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestSummary.cshtml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    ViewData["Title"] = "Type of Request";
+}
+<link rel="stylesheet" href="@Url.Content("~/css/support/support.css")" asp-append-version="true">
+@if (errorHasOccurred)
+{
+    <vc:error-summary order-of-property-names="@(new[] { nameof(Model.RequestSubject),nameof(Model.RequestDescription) })" />
+}
+<div class=" nhsuk-u-margin-top-1">
+    <a class="nhsuk-button nhsuk-button--secondary" role="button" asp-controller="RequestSupportTicket" asp-action="TypeofRequest">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestSummary.cshtml:19]: These Back links should use the [NHS Design System back-link component](https://service-manual.nhs.uk/design-system/components/back-link), as per wireframes.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SuccessPage.cshtml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+}
+<div class="nhsuk-u-margin-bottom-8">
+    <h1 class="nhsuk-heading-xl">Request @Model.TicketId submitted</h1>
+    <p class="nhsuk-body-l nhsuk-u-margin-bottom-6">
+        Your request number is @Model.TicketId. This will appear in any emails relating to the request. You should also quote it in any communication with the support team.
+            </p>
+    <p class="nhsuk-body-l nhsuk-u-margin-bottom-6">
+                You will receive an email when a member of the support team responds to it
+            </p>
+    <a class="nhsuk-button nhsuk-button--secondary" role="button" asp-controller="RequestSupportTicket" asp-action="Index">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SuccessPage.cshtml:15]: These Back links should use the [NHS Design System back-link component](https://service-manual.nhs.uk/design-system/components/back-link), as per wireframes.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml [hunk 7]
```diff
@@ -0,0 +1,88 @@
+﻿@using DigitalLearningSolutions.Data.Models.Support
+@using DigitalLearningSolutions.Web.Models.Enums
+@using DigitalLearningSolutions.Web.ViewModels.Support.RequestSupportTicket;
+@model SupportSummaryViewModel
+@{
+    var errorHasOccurred = !ViewData.ModelState.IsValid;
+}
+<a class="nhsuk-button nhsuk-button--secondary" role="button" asp-controller="RequestSupportTicket" asp-action="RequestAttachment">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml:8]: These Back links should use the [NHS Design System back-link component](https://service-manual.nhs.uk/design-system/components/back-link), as per wireframes.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/TypeOfRequest.cshtml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    var count = @Model.RequestTypes.Count();
+    var seperator = 0;
+}
+<link rel="stylesheet" href="@Url.Content("~/css/support/support.css")" asp-append-version="true">
+@if (errorHasOccurred)
+{
+    <vc:error-summary order-of-property-names="@(new[] { nameof(Model.Type) })" />
+}
+<div class=" nhsuk-u-margin-top-1">
+    <a class="nhsuk-button nhsuk-button--secondary" role="button" asp-controller="RequestSupportTicket" asp-action="Index">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/TypeOfRequest.cshtml:17]: These Back links should use the [NHS Design System back-link component](https://service-manual.nhs.uk/design-system/components/back-link), as per wireframes.

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
This is looking good. Noticed that we are using a secondary button instead of a bak-link component in the views. I've highlighted where these need changing in code comments.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml (hunk 4 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestAttachment.cshtml]: This has been implemented and changes are pushed. Please verify.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestSummary.cshtml (hunk 5 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/RequestSummary.cshtml]: This has been implemented and changes are pushed. Please verify.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SuccessPage.cshtml (hunk 6 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SuccessPage.cshtml]: This has been implemented and changes are pushed. Please verify.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml (hunk 7 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/SupportTicketSummaryPage.cshtml]: This has been implemented and changes are pushed. Please verify.

File: DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/TypeOfRequest.cshtml (hunk 8 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Web/Views/Support/RequestSupportTicket/TypeOfRequest.cshtml]: This has been implemented and changes are pushed. Please verify.

File: DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs (hunk 1 above)
Comment (sarathlal-sarangadharan) [reply DigitalLearningSolutions.Data/DataServices/RequestSupportTicketDataService.cs:28]: @rshrirohit  Regarding 6th point: **_Validation on file size is not working properly_**, this is a general issue . I could see same issue is in Delegate bulk upload page. If we try to upload huge file, it breaks the page irrespective of file format. I think we required a general fix for this.

**Pull request merged**